apps/pipeline-asset-sg/src/export-to-view.service.ts

```diff
@@ -0,0 +1,334 @@
+        titlePublic: true,
+        isNatRel: true,
+        receiptDate: true,
+        url: true,
+        locationAnalog: true,
+        lastProcessedDate: true,
+        textBody: true,
+        remark: true,
+        assetKindItemCode: true,
+        createDate: true,
+        assetFormatItemCode: true,
+        authorBiblio: true,
+        sourceProject: true,
+        description: true,
+        isExtract: true,
+        internalUseId: true,
+        publicUseId: true,
+        assetMainId: true,
+        workgroupId: true,
+
+        titleOriginal: false,
+        sgsId: false,
+        geolDataInfo: false,
+        geolContactDataInfo: false,
+        geolAuxDataInfo: false,
+        municipality: false,
+        processor: false,
+      },
+    });
+
+    const result = await this.destinationPrisma.asset.createMany({ data: assets });
+
+    log(`Created ${result.count} assets.`);
+  }
+
+  /**
+   * Export the static item tables.
+   */
+  private async exportItems() {
+    const tables = [
+      'AssetFormatItem',
+      'AssetKindItem',
+      'AutoCatLabelItem',
+      'AutoObjectCatItem',
+      'ContactKindItem',
+      'GeomQualityItem',
+      'LanguageItem',
+      'LegalDocItem',
+      'ManCatLabelItem',
+      'NatRelItem',
+      'PubChannelItem',
+      'StatusAssetUseItem',
+      'StatusWorkItem',
+    ];
+
+    for (const table of tables) {
+      await this.exportTable(table);
+    }
+  }
+
+  /**
+   * Export all entries of the table.
+   */
+  private async exportTable(table: string) {
+    const items = await this.sourcePrisma[table].findMany();
+    const result = await this.destinationPrisma[table].createMany({ data: items, skipDuplicates: true });
+
+    log(`Created ${result.count} ${table}.`);
+  }
+
+  /**
+   * Export table with ids.
+   */
+  private async export(table: string, idField = 'id', ids: number[]) {
+    const items = await this.sourcePrisma[table].findMany({ where: { [idField]: { in: ids } } });
+    const result = await this.destinationPrisma[table].createMany({ data: items, skipDuplicates: true });
+
+    log(`Created ${result.count} ${table}.`);
+  }
+
+  /**
+   * Export files.
+   * Remove files which start with 'LDoc'
+   */
+  private async exportFiles(assetIds: number[]) {
+    log(`Starting file export.`);
+    // Read all unique file ids from the assetFiles
+    const assetFiles = await this.sourcePrisma.assetFile.findMany({ where: { assetId: { in: assetIds } } });
+    const fileIds = assetFiles.map((af) => af.fileId).filter(unique);
+    const files = await this.sourcePrisma.file.findMany({ where: { fileId: { in: fileIds } } });
+
+    // Filter out files which start with 'LDoc'
+    const filesWithoutLegalDocs = files.filter((file) => !file.fileName.includes('LDoc'));
+    const fileIdsWithoutLegalDocs = filesWithoutLegalDocs.map((file) => file.fileId);
+    const assetFilesWithoutLegalDocs = assetFiles.filter((af) => fileIdsWithoutLegalDocs.includes(af.fileId));
```

Comment:
As I see it, the `"LDoc"` filter could just be moved into the initial `findMany`:
```ts
const assetFiles = await this.sourcePrisma.assetFile.findMany({
  where: {
    assetId: { in: assetIds },
    file: { fileName: { not: { contains: 'LDoc' } } },
  },
});
```